### PR TITLE
Ports RCD menu to TGUI

### PR DESCRIPTION
**Pull request by ThatLing**
# Document the changes in your pull request

Ports https://github.com/tgstation/tgstation/pull/71710. It removes the old radial menu for the RCD which sucks and replaces it with a modern TGUI window.

image
image
image

# Changelog

:cl: Synclt21
add: TGUI for RCD Just like Rapid Pipe Dispenser
code: silo link upgrades for both RLD & Plumbing RCD
del: Switch Mode, Window Glass, Airlock action icons for construction console
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
radial menu was popping off though, seems like a downgrade

**Comment by ThatLing** [comment]
Radial menus suck, I sure love cycling through 7 menus to make a single airlock with correct access

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (ToasterBiome) [inline _maps/_basemap.dm]: society

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
if we’re gonna do this:
put deconstruct back on top, it was a separate action from construction

also can we get a tooltip on hover of what things cost?

**Comment by ToasterBiome** [comment]
> if we’re gonna do this: put deconstruct back on top, it was a separate action from construction
> 
> also can we get a tooltip on hover of what things cost? deconstruct might be variously but airlocks and stuff should be workable

**Comment by ThatLing** [comment]
> if we’re gonna do this: put deconstruct back on top, it was a separate action from construction
> 
> also can we get a tooltip on hover of what things cost?

Making a seperate category with just "Deconstruct" is ugly

There is no easy way to get the cost atm, it would require a refactor which is out-of-scope for this PR

File: code/game/machinery/doors/airlock_electronics.dm [hunk 2]
```diff
@@ -49,40 +54,46 @@
 
 	return data
 
-/obj/item/electronics/airlock/ui_act(action, params)
-	if(..())
-		return
+///shared by rcd & airlock electronics
+/obj/item/electronics/airlock/proc/do_action(action, params)
+	log_world("do_action: [action]")
```
Comment (ReddicusDragon) [inline code/game/machinery/doors/airlock_electronics.dm:59]: Do we really need log_world here?

**Review by ReddicusDragon — COMMENTED** [review]
Only thing I have a possible issue with is the possibility of having a silo link. I can foresee this becoming an issue with everyone trying to run around with silo link'd rcds

**Comment by MajManatee** [comment]
I actually really like the little circle for RCDing. Quick and easy, this seems like it becomes a whole ordeal.

**Comment by ThatLing** [comment]
> I actually really like the little circle for RCDing. Quick and easy, this seems like it becomes a whole ordeal.

You think you do, but you don't

**Comment by iamthesilly** [comment]
I like the radial menu a lot better since it's simple and you can actually continue moving around and stuff in game while you're using it, though a tgui menu for selecting airlock style and access would be nice.

**Comment by Bibby0110** [comment]
would be cool if it was a preference :)

**Pull request closed without merging**